tooling/nargo_fmt/src/lib.rs

```diff
@@ -4,6 +4,10 @@
 #![warn(clippy::semicolon_if_nothing_returned)]
 #![warn(unused_qualifications, clippy::use_self)]
 
+// False positives for dev-dependencies with unused_crate_dependencies lint.
+#[cfg(test)]
+use similar_asserts as _;
```

Comment:
I don't think this is necessary. I can't get this warning to trigger on codespaces.

Reply:
```
❯ cargo test
+ command cargo test
   Compiling nargo_fmt v0.16.0 (nargo_fmt)
warning: external crate `similar_asserts` unused in `nargo_fmt`: remove the dependency or add `use similar_asserts as _;`
  |
note: the lint level is defined here
 --> tooling/nargo_fmt/src/lib.rs:2:9
  |
2 | #![warn(unused_crate_dependencies, unused_extern_crates)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: `nargo_fmt` (lib test) generated 1 warning
```

Reply:
`#![cfg_attr(not(test), warn(unused_crate_dependencies, unused_extern_crates))]` This will fix this issue and is the lint we use in all of our crates.